### ISSUE TITLE
20.04: Telegram shown twice (flatpak & deb are separate)

### ISSUE THREAD

**Issue by leviport** [pop-os/shop#157]
**Distribution (run `cat /etc/os-release`):**
Pop!_OS 20.04 
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
```
pop-shop:
  Installed: 3.1.1pop0~1584028521~20.04~f6986ab
  Candidate: 3.1.1pop0~1584028521~20.04~f6986ab
  Version table:
 *** 3.1.1pop0~1584028521~20.04~f6986ab 1002
       1002 http://apt.pop-os.org/staging/master focal/main amd64 Packages
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Issue/Bug Description:**
Telegram from Flathub and the Telegram .deb are shown as separate applications in Pop!_Shop
![Screenshot from 2020-03-16 08-22-04](https://user-images.githubusercontent.com/13512097/76768767-17fa7500-6761-11ea-9a75-8f0e64d0157f.png)

**Steps to reproduce (if you know):**
Search Telegram in Pop!_Shop

**Expected behavior:**
Telegram should show up once, and the .deb and flathub sources should be shown in the selector on the application page.

**Other Notes:**

**Comment by WatchMkr** [comment]
There are a lot of instances of this. Klavaro is another. I don't think we can reasonably handle these cases. Should we change Pop!_Shop to show the default source in the application list?

**Comment by leviport** [comment]
You mean only show the flatpak in the list if a .deb version does not exist? Well it might hide a flatpak version in favor of an older version in the repos, and users may desire the newer flatpak version. 

What are your thoughts on showing both like it does now, but showing a little flatpak icon next to any listings that have a flatpak source available? Then at least the above scenario would make more sense at first glance, and clicking on it will say whether it's a .deb or a flatpak.

**Comment by WatchMkr** [comment]
Showing both and including the source (Pop, Flathub, Ubuntu).

I just noticed Steam is showing up twice and neither is from our Pop repo. Perhaps we should close this and discuss Flathub/Pop repo cleanup on this issue: https://github.com/pop-os/repo-proprietary/issues/19

What do you think?

**Comment by leviport** [comment]
Showing both and the source sounds best to me. That's kinda what I meant with the flatpak logo, but listing them sounds more self-explanatory as long as we have room to do so.

I'm good with moving our discussion, though I think some doubled up apps won't be in our Pop repo. We'll likely still run into the same situation somewhere down the line, especially if users start installing multiple flatpak repos and PPA's.

**Comment by isantop** [comment]
A bit of relevant discussion in the AppCenter repo too: https://github.com/elementary/appcenter/issues/1011

**Comment by WuerfelDev** [comment]
I don't know if you already changed anything because it only shows one "Telegram Desktop". (So I guess this issue could be closed)
But.. the version there is 1.8.8 which is almost 6 months old: https://desktop.telegram.org/changelog
Was the wrong package selected or where is that problem coming from?

**Comment by leviport** [comment]
You're right - Telegram now only has a single entry and shows both the .deb source and the flathub source in the dropdown on the app page. I'm still seeing 2 entries for Klavaro though, so I'm not ready to close it quite yet.

**Comment by leviport** [comment]
Ah, 19.10 may be the reason. We don't plan to roll flatpak integration into 19.10, so we're only testing the new features on 20.04. We don't have isos to share yet, but you can use `pop-upgrade release upgrade systemd -f` if you really want to run it early. Just keep in mind this is pre-release so you might run into some bugs.

**Comment by ghisvail** [comment]
> Telegram now only has a single entry and shows both the .deb source and the flathub source in the dropdown on the app page.

I see only the `.deb` source, although flatpak is installed and Flathub is setup.

**Comment by jacobgkau** [comment]
A customer pointed out that FocusWriter is showing duplicate/separate search results just like Klavaro.

**Comment by WuerfelDev** [comment]
> A customer pointed out that FocusWriter and Firefox are both showing duplicate/separate search results just like Klavaro.

Confirm FocusWriter (flatuhub 1.7.5 and ubuntu-deb 1.7.4)
With Firefox both have different logos, flathub (76.0.1) and pop-deb (no version visible)


Let me just add
- Clementine Music player (flathub up-to-date, ubuntu-deb is from 2016)
- ghostwriter
- MyPaint
- neovim
- Kid3
- Ex Falso
- MediaInfo
- DOSBox & DOSBox Emulator
- FMIT
- Fuse & Fuse (GTK+) & Fuse (SDL)
- LMMS
- MuseScore
- PulseAudio Volume Control (3 times, flathub and twice ubuntu-deb)
- Ri-li
- "Sound"
- VMPK
- darktable
- Entangle
- FreeCAD
- Geeqie
- Inkscape
- LeoCAD (slightly different icons)
- nomacs & "nomacs - Image Lounge"
- Pencil 2D
- Pinta
- Scan Tailor (flathub, from 2016) & ScanTailor Advanced (flathub, from 2018)
- Scribus
- Shotwell
- Sweet Home 3D
- Synfig Studio

and more...

**Comment by Oymate** [comment]
Blender too.

**Comment by fealv** [comment]
I found this separation again today but with .deb packages from Pop_os repo
![image](https://user-images.githubusercontent.com/60493079/94328209-eae81400-ff86-11ea-935a-102c362081fc.png)
both DBeaver are from Pop_os/flathub repos and both Chromium are from Pop_os repo. It's not an entry for each one, but the same options twice.

**Comment by ArikRahman** [comment]
+1, Steam, Godot, Vim, Neovim, Arduino IDE, DavMail, Filezilla, nheko, Remote Desktop Manager, Marble, Endless Sky, Viking. Neverball, etc. all have duplicates for example. It was confusing as a first time user, not sure if I was installing the right one or not.
